src/tree/SubscriptionTreeItem.ts

```diff
@@ -92,48 +92,29 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
-                }
-            } else if (!this._triggeredByDefaultGroupBySetting) {
-                if (clearCache) {
-                    this.resetCache();
-                }
-
-                if (this.cache.appResources.length === 0) {
-                    const resources = await applicationResourceProviders[azureResourceProviderId]?.provideResources(this.subscription) ?? [];
-
-                    // To support multiple app resource providers, need to use this pattern
-                    // await Promise.all(applicationResourceProviders.map((provider: ApplicationResourceProvider) => async () => this.rgsItem.push(...(await provider.provideResources(this.subscription) ?? []))));
-
-                    resources.forEach(item => ext.activationManager.onNodeTypeFetched(item.type));
-                    this.cache.appResources = resources.map((resource: AppResource) => AppResourceTreeItem.Create(this, resource));
-                }
-                await this.createTreeMaps(clearCache, context);
-
-                // on first load, check if there was persistent setting
-                focusGroupTreeItem = await this.tryGetFocusGroupTreeItem(focusedGroupId);
-                if (focusGroupTreeItem) {
-                    return [focusGroupTreeItem];
-                }
-            }
+        if (clearCache) {
+            this.resetCache();
+        }
 
-            let groupTreeItems = Object.values(this.treeMap) as GroupTreeItemBase[];
-            if (!settingUtils.getWorkspaceSetting('showHiddenTypes') &&
-                settingUtils.getWorkspaceSetting<string>('groupBy') === GroupBySettings.ResourceType) {
-                groupTreeItems = groupTreeItems.filter(ti => this._azExtGroupConfigs.some(config => config.id === ti.config.id));
-            }
+        if (this.cache.appResources.length === 0) {
+            const resources = await applicationResourceProviders[azureResourceProviderId]?.provideResources(this.subscription) ?? [];
 
-            return groupTreeItems;
-        } finally {
-            this._triggeredByDefaultGroupBySetting = false;
-            this._triggeredByFocusGroupSetting = false;
+            // To support multiple app resource providers, need to use this pattern
+            // await Promise.all(applicationResourceProviders.map((provider: ApplicationResourceProvider) => async () => this.rgsItem.push(...(await provider.provideResources(this.subscription) ?? []))));
+
+            resources.forEach(item => ext.activationManager.onNodeTypeFetched(item.type));
+            this.cache.appResources = resources.map((resource: AppResource) => AppResourceTreeItem.Create(this, resource));
+        }
+        await this.createTreeMaps(clearCache, context);
+
+        const focusedGroupId = await ext.context.workspaceState.get('focusedGroup') as string;
+        // on first load, check if there was persistent setting
+        const focusGroupTreeItem = await this.tryGetFocusGroupTreeItem(focusedGroupId);
```

Comment:
Can we have `tryGetFocusGroupTreeItem` internally take care of looking up the setting value?